PSReadLine/Render.cs

```diff
@@ -340,135 +349,322 @@ void MaybeEmphasize(int i, string currColor)
             return currentLogicalLine + 1;
         }
 
-        private void ReallyRender(RenderData renderData, string defaultColor)
+        /// <summary>
+        /// Flip the color on the prompt if the error state changed.
+        /// </summary>
+        /// <returns>
+        /// A bool value indicating whether we need to flip the color,
+        /// namely whether we moved cursor to the initial position.
+        /// </returns>
+        private bool RenderErrorPrompt(RenderData renderData, string defaultColor)
         {
-            string activeColor = "";
+            // We may need to flip the color on the prompt if the error state changed.
 
-            void UpdateColorsIfNecessary(string newColor)
+            var bufferWidth = _console.BufferWidth;
```

Comment:
Wouldn't use `var` here or below, since I can't tell reading in the review what types the variables are

Reply:
Updated to use the type explicitly.